test/graphkb.test.js

```diff
@@ -1,4 +1,108 @@
+    simplifyRecordsLinks,
+} = require('../src/graphkb');
+
+describe('orderPreferredOntologyTerms', () => {
+    test('prefer non-deprecated', () => {
+        expect(orderPreferredOntologyTerms(
+            { deprecated: true }, { deprecated: false },
+        )).toBe(1);
+        expect(orderPreferredOntologyTerms(
+            { deprecated: false }, { deprecated: true },
+        )).toBe(-1);
+    });
+
+    test('prefer terms with independent sourceId', () => {
+        expect(orderPreferredOntologyTerms(
+            { alias: false }, { alias: true },
+        )).toBe(-1);
+        expect(orderPreferredOntologyTerms(
+            { alias: true }, { alias: false },
+        )).toBe(1);
+        expect(orderPreferredOntologyTerms(
+            { }, { dependency: true },
+        )).toBe(-1);
+        expect(orderPreferredOntologyTerms(
+            { dependency: true }, { },
+        )).toBe(1);
+    });
+
+    test('prefer generic to versioned terms', () => {
+        expect(orderPreferredOntologyTerms(
+            { }, { sourceIdVersion: '' },
+        )).toBe(-1);
+        expect(orderPreferredOntologyTerms(
+            { sourceIdVersion: '' }, { },
+        )).toBe(1);
+    });
+
+    test('prefer newer version of same record', () => {
+        expect(orderPreferredOntologyTerms(
+            { sourceIdVersion: '2019-10-08' }, { sourceIdVersion: '2020-10-08' },
+        )).toBe(-1);
+        expect(orderPreferredOntologyTerms(
+            { sourceIdVersion: '2020-10-08' }, { sourceIdVersion: '2019-10-08' },
+        )).toBe(1);
+    });
+
+    test('prefer newer source version of same record', () => {
+        expect(orderPreferredOntologyTerms(
+            { source: { version: 1 }, sourceIdVersion: '2019-10-08' },
+            { source: { version: 2 }, sourceIdVersion: '2019-10-08' },
+        )).toBe(-1);
+        expect(orderPreferredOntologyTerms(
+            { source: { version: 2 }, sourceIdVersion: '2019-10-08' },
+            { source: { version: 1 }, sourceIdVersion: '2019-10-08' },
+        )).toBe(1);
+    });
+
+    test('prefer terms with descriptions', () => {
+        expect(orderPreferredOntologyTerms(
+            { description: 'a description', sourceIdVersion: '2019-10-08' },
+            { description: '', sourceIdVersion: '2019-10-08' },
+        )).toBe(-1);
+        expect(orderPreferredOntologyTerms(
+            { description: '', sourceIdVersion: '2019-10-08' },
+            { description: 'a description', sourceIdVersion: '2019-10-08' },
+        )).toBe(1);
+    });
+
+    test('use source rank to sort results', () => {
+        expect(orderPreferredOntologyTerms(
+            { source: { sort: 1 }, sourceId: 1 },
+            { source: { sort: 2 }, sourceId: 2 },
+        )).toBe(-1);
+        expect(orderPreferredOntologyTerms(
+            { source: { sort: 2 }, sourceId: 1 },
+            { source: { sort: 1 }, sourceId: 2 },
+        )).toBe(1);
+        expect(orderPreferredOntologyTerms(
```

Comment:
are these in the wrong test block? only the first 2 expects appear to have the sort property

Reply:
Not sure to understand the question.
Lines 73-80 of ./test/graphkb.test.js are intended to test lines 181-184 of ./src/graphkb.js

Reply:
Not a blocker, I just meant that the title 'use source rank to sort results' implies testing the rank/sort field of source so you might want the other tests in a different test block with a more descriptive name. Or remove "rank" from the name.

Reply:
Ok, I see. I just blindly follow the [code](https://github.com/bcgsc/pori_graphkb_loader/blob/007cab68a87f84c47a38c98452dd0b67a56ec3ad/src/graphkb.js#L180)  structure without questionning (Look, I used your trick!)